internal/broker/export_test.go

```diff
@@ -2,13 +2,39 @@ package broker
 
 import (
 	"context"
-	"fmt"
-	"os"
-	"path/filepath"
 
 	"github.com/ubuntu/authd-oidc-brokers/internal/providers/info"
+	tokenPkg "github.com/ubuntu/authd-oidc-brokers/internal/token"
 )
 
+func (cfg *Config) SetClientID(clientID string) *Config {
+	cfg.clientID = clientID
+	return cfg
+}
+
+func (cfg *Config) SetIssuerURL(issuerURL string) *Config {
+	cfg.issuerURL = issuerURL
+	return cfg
+}
+
+func (cfg *Config) SetHomeBaseDir(homeBaseDir string) *Config {
+	cfg.homeBaseDir = homeBaseDir
+	return cfg
+}
+
+func (cfg *Config) SetAllowedSSHSuffixes(allowedSSHSuffixes []string) *Config {
+	cfg.allowedSSHSuffixes = allowedSSHSuffixes
+	return cfg
+}
+
+func (cfg *Config) ClientID() string {
+	return cfg.clientID
+}
+
+func (cfg *Config) IssuerURL() string {
+	return cfg.issuerURL
+}
+
```

Comment:
In Go, we tend to drop this kind of builder pattern for the variadic arguments approach. You can see an example here of how we do it: https://github.com/ubuntu/authd/blob/223a072ec87fbd432c3f31861e684042dfe1a941/internal/testutils/golden.go#L32.

Reply:
If I understand correctly, that requires adding a function `NewConfig(opts ...func(*Config)) *Config` and functions like:

```
func WithClientID(clientID string) func(*Config) {
	return func(cfg *Config) {
		cfg.clientID = clientID
	}
}
```

which can then be passed to `NewConfig`, right? That doesn't allow setting fields after initialization of the struct, as we do [here]( https://github.com/ubuntu/authd-oidc-brokers/blob/dd8e0c086ea07c470dd6d0f04499db0369eebdfa/internal/broker/helper_test.go#L35-L37) though, right?

Reply:
Also, I have the impression that the setters I added follow the pattern of the other functions defined in the same file more closely, for example [`SetAuthInfo`](https://github.com/ubuntu/authd-oidc-brokers/blob/dd8e0c086ea07c470dd6d0f04499db0369eebdfa/internal/broker/export_test.go#L97-L97) and [`SetAvailableMode`](https://github.com/ubuntu/authd-oidc-brokers/blob/dd8e0c086ea07c470dd6d0f04499db0369eebdfa/internal/broker/export_test.go#L111-L111).

Reply:
> That doesn't allow setting fields after initialization of the struct, as we do here though, right? 

It definitely doesn't. However, you could create a new object based on the current one (to set the field that you need) Ps.: This is also subject to what we decide regarding the config structs.

> Also, I have the impression that the setters I added follow the pattern of the other functions defined in the same file more closely, for example SetAuthInfo and SetAvailableMode.

In this case, the sets were due to having to alter an existing (complex) object (which, yeah, I know we do that currently for the config also [when looking at the example you've mentioned] but it could be avoided, no?).

Reply:
Since my change follows the pattern in that file, can we keep it as is in this PR and if we want to use a different pattern in that file, do that in a separate PR?

Reply:
Even though the names are similar, it is not the same pattern. The ones you've mentioned only alter the object instead of returning it (like you did here). That's what I meant by "builder pattern". If you need something like this, it's better to follow the convention and use the optional arguments instead (the whole `WithXX` syntax I pointed to).

I believe we have three ways of handling this in a Go-like way:

1) Create a new Config struct with the exposed fields in `export_test.go` and remap it to the new config file (the worse option);
2) Use the `SetXX`s, but as they are meant to be used: altering the config object itself without returning anything (the functions you've mentioned are named SetXX but are not Setters like the ones you did here) (good option and require less refactoring);
3) Use optional arguments like I've mentioned (IMO, best option although it will require more work to implement);